testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/SdJwtSigningServiceTest.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    @Test
+    public void testRsaSignedCredentialWithKeyId() {
+        getTestingClient()
+                .server(TEST_REALM_NAME)
+                .run(session ->
+                        testSignSDJwtCredential(
+                                session,
+                                Optional.of("did:web:test.org#key-id"),
+                                Algorithm.RS256,
+                                Map.of("id", String.format("uri:uuid:%s", UUID.randomUUID()),
+                                        "test", "test",
+                                        "arrayClaim", List.of("a", "b", "c")),
+                                0,
+                                List.of()));
+    }
+
+    @Test
+    public void testRsaSignedCredentialWithoutAdditionalClaims() {
+        getTestingClient()
+                .server(TEST_REALM_NAME)
+                .run(session ->
+                        testSignSDJwtCredential(
+                                session,
+                                Optional.empty(),
+                                Algorithm.RS256,
+                                Map.of(),
+                                0,
+                                List.of()));
+    }
+
+    public static void testSignSDJwtCredential(KeycloakSession session, Optional<String> keyId, String
+            algorithm, Map<String, Object> claims, int decoys, List<String> visibleClaims) {
+        KeyWrapper keyWrapper = getKeyFromSession(session);
+
+        SdJwtSigningService signingService = new SdJwtSigningService(
+                session,
+                new ObjectMapper(),
+                keyWrapper.getKid(),
+                algorithm,
+                "vc+sd-jwt",
+                "sha-256",
+                "did:web:test.org",
+                decoys,
+                visibleClaims,
+                new StaticTimeProvider(1000),
+                keyId);
+
+        VerifiableCredential testCredential = getTestCredential(claims);
+
+        String sdJwt = signingService.signCredential(testCredential);
+        SignatureVerifierContext verifierContext = null;
+        switch (algorithm) {
+            case Algorithm.ES256: {
+                verifierContext = new ServerECDSASignatureVerifierContext(keyWrapper);
+                break;
+            }
+            case Algorithm.RS256: {
+                verifierContext = new AsymmetricSignatureVerifierContext(keyWrapper);
+                break;
+            }
+            default: {
+                fail("Algorithm not supported.");
+            }
+        }
+        // the sd-jwt is dot-concatenated header.payload.signature~disclosure1~___~disclosureN
+        // we first split the disclosuers
+        String[] splittedSdToken = sdJwt.split("~");
+        // and then split the actual token part
+        String[] splittedToken = splittedSdToken[0].split("\\.");
+
+        String jwt = new StringJoiner(".")
+                // header
+                .add(splittedToken[0])
```

Comment:
According to https://drafts.oauth.net/oauth-sd-jwt-vc/draft-ietf-oauth-sd-jwt-vc.html#name-jose-header, "typ" must be "vc+sd-jwt". How about checking it somewhere?